mkdocs_section_index/plugin.py

```diff
@@ -24,7 +28,16 @@ def on_nav(self, nav: Navigation, config, files) -> Navigation:
                 if not isinstance(section, Section) or not section.children:
                     continue
                 todo.append(section.children)
-                page = section.children[0]
+                name = self.config['use-page']
+                if name is None:
+                    page = section.children[0]
+                else:
+                    for child in section.children:
+                        if child.file.name == name:
```

Comment:
Tip: to implement this as file-based, I believe you'd use `os.path.basename(child.file.src_path)`